src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -6,16 +6,97 @@
     from the schema manifest defined by RAD.
 """
 
+import pathlib
+
 import asdf
+import numpy as np
 from astropy.table import QTable
 
 from roman_datamodels import stnode
 
+from ..stnode import DNode
 from ._core import DataModel
 from ._utils import _node_update
 
 __all__ = []
 
+DTYPE_MAP = {}
+
+
+class _ParquetMixin:
+    """Gives SourceCatalogModels the ability to save to parquet files."""
+
+    def to_parquet(self, file=None, fdir=None):
```

Comment:
```suggestion
    def to_parquet(self, filepath):
```
Can we simplify this a bit to just accept a path?

Reply:
I think so. I'll do that now.